tooling/nargo_cli/src/cli/fmt_cmd.rs

```diff
@@ -33,7 +33,11 @@ pub(crate) fn run(_args: FormatCommand, config: NargoConfig) -> Result<(), CliEr
             let file_id = file_manager.add_file(&entry.path()).expect("file exists");
             let (parsed_module, errors) = parse_file(&file_manager, file_id);
 
-            if !errors.is_empty() {
+            let is_warnings = errors
+                .iter()
+                .all(|err| matches!(err.reason(), Some(ParserErrorReason::ExperimentalFeature(_))));
```

Comment:
Lets rename this to `is_all_warnings`.

Manually matching against a single error variant here seems like it'd be easy to break when another parser error is added. Can we replace the match here with a `is_warning` method on parser errors instead?